src/bg/user-script-detect.js

```diff
@@ -0,0 +1,103 @@
+/*
+Add listeners to detecting user scripts and opening the installation
+dialog.
+*/
+
+
+(function() {
+
+const userScriptTypes = new Set([
+   'text/plain',
+   'application/ecmascript',
+   'application/javascript',
+   'application/x-javascript',
+   'text/ecmascript',
+   'text/javascript',
+]);
+
+
+// Examine headers before determining if script checking is needed
+function checkHeaders(responseHeaders) {
+  for (header of responseHeaders) {
+    if ('Content-Type' === header.name && userScriptTypes.has(header.value)) {
```

Comment:
These should be case-insensitive matches, as the server is not guaranteed to provide any particular case for these values.  (Just tested, when I send all-lower values, the header name is exactly "content-type" here.)